catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -929,10 +953,32 @@ private CswRecordCollection queryCsw(GetRecordsType request) throws CswException
             try {
                 queryRequest = queryFactory.updateQueryRequestTags(queryRequest,
                         request.getOutputSchema());
-                LOGGER.debug("Attempting to execute query: {}", queryRequest);
-                QueryResponse queryResponse = framework.query(queryRequest);
+
+                LOGGER.debug("Attempting to execute paged query: {}", queryRequest);
+                AtomicLong hitCount = new AtomicLong(0);
+
+                QueryFunction qf = qr -> {
+                    SourceResponse sr = framework.query(qr);
+                    hitCount.compareAndSet(0, sr.getHits());
+                    return sr;
+                };
+
+                ResultIterable results = ResultIterable.resultIterable(qf,
+                        queryRequest,
+                        request.getMaxRecords()
+                                .intValue());
+                List<Result> resultList = results.stream()
+                        .collect(Collectors.toList());
+
+                long totalHits = hitCount.get();
+                totalHits = totalHits != 0 ? totalHits : resultList.size();
+
+                QueryResponse queryResponse = new QueryResponseImpl(queryRequest,
+                        resultList,
+                        totalHits);
```

Comment:
Ah! Make sense now. @rzwiefel is right. This would be a great time to add an inline comment to explain _why_ we're using `totalHits` instead of `resultList.size()`.